crowd_anki/importer/import_dialog.py

```diff
@@ -0,0 +1,146 @@
+
+
+class ImportDialog(QDialog):
+    def __init__(self, deck_json, config, parent=None):
+        super().__init__(None)
+        self.parent = parent
+        self.form = ConfigUI()
+        self.form.setupUi(self)
+        self.userConfig = ConfigSettings.get_instance()
+        self.deck_json = deck_json
+        self.import_settings = ImportJsonSetting(config)
+        self.ui_initial_setup()
+
+        self.final_import_config: ImportConfig = None
+
+    def accept(self):
+        self.get_import_config()
+        super().accept()
+
+    def reject(self):
+        self.get_import_config()
+        super().reject()
+
+    def ui_initial_setup(self):
+        self.setup_personal_field_selection()
+        self.setup_misc()
+
+    def setup_personal_field_selection(self):
+        heading_font = QFont()
+        heading_font.setBold(True)
+        heading_font.setUnderline(True)
+
+        def add_header(name):
+            heading_ui = QListWidgetItem(name)
+            heading_ui.setFlags(Qt.ItemIsEnabled)
+            heading_ui.setSizeHint(QSize(self.form.list_personal_fields.width(), 30))
+            heading_ui.setFont(heading_font)
+            self.form.list_personal_fields.addItem(heading_ui)
+
+        def add_field(name, is_personal):
+            field_ui = QListWidgetItem(name)
+            field_ui.setCheckState(Qt.Checked if is_personal else Qt.Unchecked)
+            field_ui.setFlags(Qt.ItemIsEnabled | Qt.ItemIsUserCheckable)
+            self.form.list_personal_fields.addItem(field_ui)
+
+        for model in self.deck_json["note_models"]:
+            model_name = model["name"]
+            add_header(model_name)
+
+            for field in model["flds"]:
+                field_name = field["name"]
+                add_field(field_name, (model_name, field_name) in self.import_settings.personal_fields)
+
+    def setup_misc(self):
+        self.form.import_message_textbox.setText(self.import_settings.import_message)
+
+        if self.import_settings.suggest_tag_imported_cards:
+            self.form.cb_tag_cards.setCheckState(Qt.Checked)
+            self.form.cb_tag_cards.setText("Tag Cards (Suggested by Deck Manager!)")
+        # else:
+        # set as default from config settings
+
+        if self.userConfig.import_notes_ignore_deck_movement:
+            self.form.cb_ignore_move_cards.setCheckState(Qt.Checked)
+
+        self.form.cb_headers.setCheckState(Qt.Checked)
+        self.form.cb_note_models.setCheckState(Qt.Checked)
+        self.form.cb_notes.setCheckState(Qt.Checked)
+        self.form.cb_media.setCheckState(Qt.Checked)
+
+        # TODO: Deck Parts to Use
+
+    def get_import_config(self):
+        config = ImportConfig()
+
+        config.personal_fields = []
+        current_note_model = ""
+        fields: List[QListWidgetItem] = [self.form.list_personal_fields.item(i) for i in
+                                         range(self.form.list_personal_fields.count())]
+        for field in fields:
```

Comment:
maybe split things into different sub-panels instead of having a continuous list of heterogeneous  widgets?